packages/nvidia_nat_core/tests/nat/middleware/test_circuit_breaker_middleware.py

```
@@ -0,0 +1,552 @@
+            cooldown_period=10.0,
+            circuit_breaker_message="Please retry later.",
+        )
+
+        call_next = AsyncMock(side_effect=RuntimeError("Failure"))
+        with pytest.raises(RuntimeError):
+            await middleware.function_middleware_invoke("input", call_next=call_next, context=function_context)
+        assert middleware.get_state("test_tool") == CircuitBreakerState.OPEN
+
+        call_next.reset_mock()
+        with pytest.raises(CircuitBreakerOpenError,
+                           match="Circuit breaker is OPEN for 'test_tool'.*Please retry later."):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)